rpc/common.go

```diff
@@ -133,36 +133,42 @@ func GetGroupControllerCapabilities(ctx context.Context, conn *grpc.ClientConn)
 
 // ProbeForever calls Probe() of a CSI driver and waits until the driver becomes ready.
 // Any error other than timeout is returned.
-func ProbeForever(conn *grpc.ClientConn, singleProbeTimeout time.Duration) error {
+func ProbeForever(ctx context.Context, conn *grpc.ClientConn, singleProbeTimeout time.Duration) error {
+	logger := klog.FromContext(ctx)
+	ticker := time.NewTicker(probeInterval)
+
 	for {
-		klog.Info("Probing CSI driver for readiness")
-		ready, err := probeOnce(conn, singleProbeTimeout)
-		if err != nil {
-			st, ok := status.FromError(err)
-			if !ok {
-				// This is not gRPC error. The probe must have failed before gRPC
-				// method was called, otherwise we would get gRPC error.
-				return fmt.Errorf("CSI driver probe failed: %s", err)
-			}
-			if st.Code() != codes.DeadlineExceeded {
-				return fmt.Errorf("CSI driver probe failed: %s", err)
-			}
-			// Timeout -> driver is not ready. Fall through to sleep() below.
-			klog.Warning("CSI driver probe timed out")
-		} else {
-			if ready {
-				return nil
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
Better use this:
```suggestion
			return context.Cause(ctx)
```

If the context gets canceled using the new(is) `WithCancelCause`, then we get that cause, otherwise the normal "context canceled".